src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputNoWhitespaceBeforeAnnotations.java

```diff
@@ -0,0 +1,44 @@
+package com.google.checkstyle.test.chapter4formatting.rule462horizontalwhitespace;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+@Target(ElementType.TYPE_USE)
+@interface NonNull {
+
+}
+
+public class InputNoWhitespaceBeforeAnnotations {
+
+    public void foo(final char @NonNull [] param) {}
+
+    @NonNull int @NonNull[] @NonNull[] fiel1; // OK until #8205.
+    @NonNull int @NonNull [] @NonNull [] field2;
+    //@NonNull int @NonNull ... field3; // non-compilable
+    //@NonNull int @NonNull... field4; // non-compilable
+
+
+    public void foo1(final char[] param) { // OK until #8205.
+    }
+
+    public void foo2(final char [] param) {
+    }
+
+    public void foo3(final char @NonNull[] param) { // OK until #8205.
+    }
+
+    public void foo4(final char @NonNull [] param) {
+    }
+
+    void test1(String... param) { // OK until #8205.
+    }
+
+    void test2(String ... param) { // violation on 8.32
+    }
+
+    void test3(String @NonNull... param) { // OK until #8205.
```

Comment:
please remove dot at the end of comment.

Reply:
done